flink-ml-parent/flink-ml-lib/src/main/java/org/apache/flink/ml/common/mapper/ModelMapper.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.flink.ml.common.mapper;
+
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.ml.api.misc.param.Params;
+import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.table.types.DataType;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.Preconditions;
+
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ * Abstract class for mappers with model.
+ */
+public abstract class ModelMapper extends Mapper {
+
+	/**
+	 * Specify where to load model data.
+	 */
+	private enum ModelSourceType implements Serializable {
+		/**
+		 * Load model data from flink's broadcast dataset.
+		 */
+		BroadcastDataset,
+
+		/**
+		 * Load model data from data rows.
+		 */
+		Rows
+	}
```

Comment:
It is probably worth separating out the model source so it can be reused by other classes. Also it seems indication that we may need two ModelSource implementations. I have created a PR against your PR branch with my suggestions.

Reply:
Yes, it is better to separate out the model source.